### PR TITLE
implemented get user who worked the most hours [Date: 03/04/2023] [Ti…

### PR DESCRIPTION
**Pull request by i-api**
Tim feature get user who worked the most hours

### CODE REVIEW

File: server/src/controller/userController.js [hunk 1]
```diff
@@ -20,6 +20,33 @@ export const get = async (req, res) => {
   return res.status(StatusCodes.ACCEPTED).json({ users });
 };
 
+export const getUserWhoWorkedTheMostHours = async (req, res) => {
+  if (validate(req, res)) {
+    return res;
+  }
+  const users = await prisma.user.findMany({
+    where: {
```
Comment (ChrisWilhelm) [inline server/src/controller/userController.js:28]: you should instead use an aggregation query, and group the shifts by user id.

**Review by ChrisWilhelm — CHANGES_REQUESTED** [review]

File: server/src/controller/userController.js (hunk 1 above)
Comment (ChrisWilhelm) [reply server/src/controller/userController.js:28]: the way you did it works, this is just a more efficient method and is good to learn

Comment (i-api) [reply server/src/controller/userController.js:28]: i gave the prisma.user.aggregate() method a shot, the docs were a little confusing, need some help
https://www.prisma.io/docs/concepts/components/prisma-client/aggregation-grouping-summarizing

Comment (i-api) [reply server/src/controller/userController.js:28]: const users = await prisma.user.aggregate({
    where: {
      shiftsWorked: {
        isEmpty: false, // we only want users who have worked at least one shift
      },
    },
    orderBy: {
      _sum: {
        shiftsWorked: {
          end: { decrement: start },
        },
      },
      _sum: "asc",
    },
  });


I'm assuming this is wrong ^

Comment (ChrisWilhelm) [reply server/src/controller/userController.js:28]: got it, we can work on it monday

**Comment by i-api** [comment]
finished. Chris gave me the "ok" to go with the hashmap method.

**Pull request merged**